src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationPath.cs

```diff
@@ -54,7 +54,7 @@ public static string Combine(IEnumerable<string> pathSegments)
                 return path;
             }
 
-            int lastDelimiterIndex = path.LastIndexOf(KeyDelimiter, StringComparison.OrdinalIgnoreCase);
+            int lastDelimiterIndex = path.LastIndexOf(':');
             return lastDelimiterIndex == -1 ? path : path.Substring(lastDelimiterIndex + 1);
```

Comment:
```suggestion
            return lastDelimiterIndex < 0 ? path : path.Substring(lastDelimiterIndex + 1);
```
as micro-optimization. Comparing to `0` is a little tad cheaper at cpu-level (comparison and the jump can be fused together).